### DISCUSSION TITLE
We're trying out Dexie.js Discussions!

### DISCUSSION THREAD

**Discussion by dfahlander** [dexie/Dexie.js#1232] [General]
## 👋 Welcome!
  We're giving Github Discussions a try. Might promote it more in future. For now, feel free to experiment with it:
  * Engage with other community members.
  * Share ideas.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.
  * Ask questions you’re wondering about.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Natreve** [comment]
Hey!, I've been using Dexie.js in my current project and loving it!, I managed to do some really cool things with it like this. 

```javascript
const getUnuploadedImages = async query => {
    return await db
      .table("images")
      .where("[metadata.id+uploadProgress]")
      .anyOf(query)
      .toArray()
  }
```

Just wanted to know if there are any plans to make **Dexie.js** [tree shakable](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/tree-shaking/) in the near future. It's something interesting to look into **Gatsby** and **firebase** are two solid examples of projects that take this approach.

```javascript

import { Dexie, table, where, anyOf, toArray} from "dexie"
const getUnuploadedImages = async query => {
   const db = Dexie(config)
   const _images = await table(db, "images")
   const imagesQuery = where(_images, "[metadata.id+uploadProgress]")
   const queryResult =  anyOf(query)
   const results =  toArray(queryResult)
}
```

**Reply by dfahlander** [reply]
Thanks for sharing!

I will try to make as much as possible tree shakable in the next major version of Dexie without sacrificing backward compability too much. For example, all static methods on Dexie could be named exports which could mean a lot for the tree-shakabiility. At some point, the idea is to allow also operators to be tree shakable and let queries be expressed like this:

```js
import { between } from "dexie";

table.where({
  name: "Foo",
  age: between(20, 65)
});
```

A bonus would be partial mongo compatibility (`between(20, 65)` would just be a helper function returning `{$gte: 20, $lte 65}`).

With this approach your query could be expressed as:

```js
const getUnuploadedImages = async query => {
    return await db.table("images").where({
      "[metadata.id+uploadProgress]": anyOf(query)
    }).toArray();
}
```

**Reply by Natreve** [reply]
Nice! Keep up the great work
